openadapt/app/dashboard/api/recordings.py

```diff
@@ -73,6 +83,18 @@ async def get_recording_detail(websocket: WebSocket, recording_id: int) -> None:
                     {"type": "num_events", "value": len(action_events)}
                 )
 
+                def convert_to_str(event_dict: dict) -> dict:
+                    """Convert the keys to strings."""
+                    if "key" in event_dict:
+                        event_dict["key"] = str(event_dict["key"])
+                    if "canonical_key" in event_dict:
+                        event_dict["canonical_key"] = str(event_dict["canonical_key"])
+                    if "reducer_names" in event_dict:
+                        event_dict["reducer_names"] = list(event_dict["reducer_names"])
```

Comment:
How about:

```
for attr in ("key", "canonical_key", "reducer_names"):
    if attr in event_dict:
        ...
```

Reply:
Can you also please clarify why this is necessary? i.e. was it a json serialization error? If that is the case I wonder if it would make more sense to implement a custom serializer.